llvm-project/stage1.cmake

```diff
@@ -0,0 +1,134 @@
+# Please try to keep these cmake variables alphabetically sorted.
+
+# Enable optimizations, as opposed to a debug build.
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+
+# Use Alpine's clang and clang++ from their clang package as the stage0
+# compilers.
+set(CMAKE_CXX_COMPILER "/usr/bin/clang++" CACHE FILEPATH "")
+set(CMAKE_C_COMPILER "/usr/bin/clang" CACHE FILEPATH "")
+
+# Alpine's clang does not default to LLD; it defaults to BFD. Use LLD when
+# building executables and shared libraries.
+set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+set(CMAKE_SHARED_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+
+# Set the default target triple to match the host.
+# TODO: passing in the value of $(clang -print-multiarch) causes failures.
+# It seems that alpine clang's default target triple is x86_64-linux-gnu.
+# Perhaps missing alpine in the triple causes some incompatibility?
```

Comment:
I don't think `clang -print-multarch` prints the default target triple, it seems like we really want `clang -print-target-triple`?

On AArch64:

```
$ podman run --rm -ti docker.io/alpine:edge sh -c "apk add --no-cache clang 2>&1 >/dev/null && clang -print-target-triple"
aarch64-alpine-linux-musl
```

On x86_64:

```
$ podman run --rm -ti docker.io/alpine:edge sh -c "apk add --no-cache clang 2>&1 >/dev/null && clang -print-target-triple"
x86_64-alpine-linux-musl
```

Reply:
ah, makes sense the flag for the target triple is named as such! My mistake for trying the wrong one. Let me give that a shot. Thanks for the tip.

Reply:
Looks like the mistake I was making when testing this out locally was using the -Dvariable in a `set` statement.
ie. `set(NEWVAR "${DVAR}")`. I was missing the ${} and "".

Reply:
idk, I'm still running into issues trying to replace this. Will look further some other time.

Reply:
Are you trying to replace this within the `cmake` cache or in the Dockerfile? Happen to have a diff that you tried so I can see if I can reproduce locally?

Reply:
I'm trying to have `cmake .... -D TRIPLE=$(clang -print-target-triple)` in the Dockerfile, then `set(LLVM_DEFAULT_TARGET_TRIPLE "${TRIPLE}")` in the stageX.cmake.

Reply:
Ah yeah, I don’t think that will work. I think we just ditch trying to preserve that in the cache and just define the value during the `cmake` invocation. It is not a static value so we shouldn’t try to treat it as such.

`cmake -B … -DLLVM_DEFAULT_TARGET_TRIPLE=$(clang -print-target-triple) …`